ouroboros-consensus/src/Ouroboros/Consensus/NodeNetwork.hs

```diff
@@ -259,42 +264,72 @@ protocolCodecs cfg version = ProtocolCodecs {
           (nodeEncodeHeaderHash (Proxy @blk))
           (nodeDecodeHeaderHash (Proxy @blk))
 
-    , pcBlockFetchCodecSerialised =
-        codecBlockFetchSerialised
-          (nodeEncodeHeaderHash (Proxy @blk))
-          (nodeDecodeHeaderHash (Proxy @blk))
+    , pcBlockFetchCodecSerialised = staticBlockFetchCodecSerialised
 
-    , pcTxSubmissionCodec =
-        codecTxSubmission
-          nodeEncodeGenTxId
-          nodeDecodeGenTxId
-          nodeEncodeGenTx
-          nodeDecodeGenTx
+    , pcTxSubmissionCodec         = staticTxSubmissionCodec
 
-    , pcLocalChainSyncCodec =
-        codecChainSyncSerialised
-          (encodePoint (nodeEncodeHeaderHash (Proxy @blk)))
-          (decodePoint (nodeDecodeHeaderHash (Proxy @blk)))
-          (encodeTip   (nodeEncodeHeaderHash (Proxy @blk)))
-          (decodeTip   (nodeDecodeHeaderHash (Proxy @blk)))
+    , pcLocalChainSyncCodec       = staticLocalChainSyncCodec
 
-    , pcLocalTxSubmissionCodec =
-        codecLocalTxSubmission
-          nodeEncodeGenTx
-          nodeDecodeGenTx
-          (nodeEncodeApplyTxError (Proxy @blk))
-          (nodeDecodeApplyTxError (Proxy @blk))
+    , pcLocalTxSubmissionCodec    = staticLocalTxSubmissionCodec
 
-    , pcLocalStateQueryCodec =
-        codecLocalStateQuery
-          (encodePoint (nodeEncodeHeaderHash (Proxy @blk)))
-          (decodePoint (nodeDecodeHeaderHash (Proxy @blk)))
-          nodeEncodeQuery
-          nodeDecodeQuery
-          nodeEncodeResult
-          nodeDecodeResult
+    , pcLocalStateQueryCodec      = staticLocalStateQueryCodec
     }
 
```

Comment:
Yes, the idea is to replace the locally defined codecs in the Wallet
and cardano-rest and other places with these static-Versions of the codecs.
If in the future one of codecs becomes configurable,
the codec has to be removed from the exports.
This will break the libraries that uses this codec and the problem will be noticed.

Reply:
@mrBliss let's keep this part as simple as possible.  The requirement here is to make it as easy as possible to run and maintain client applications (good example is `cardano-cli get-tip`, it just needs to know the block type).  Since `node-to-client` protocols don't depend on `TopLevelConfig` let's not include it.  Do you see some dependency in forcible future? Will the `LocalStateQuery` or `LocalTxMonitor` protocols depend on parts of `TopLevelConfig`?  If so I think it's good to start with no dependency, and then add only the necessary dependencies. This will make us think how to properly do it when we will need it.

@MarcFontaine these codes should be versioned by `NodeToClientVersion`.  There's only one version now, but still it's better to put that already in place. I guess, they no longer become static.

Reply:
I guess it's fine for now. But as soon as they *do* start depending on the `TopLevelConfig`, I don't want to be blocked by these `static*` definitions.

Reply:
I actually want us to be blocked by them - this will give us moment to think how to make it as simple as possible for the clients as well.